core/modules/replica/DatabaseMySQLRow.cc

```diff
@@ -0,0 +1,174 @@
+bool Row::get(std::string const& columnName, uint32_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, uint16_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, uint8_t&  value) const { return ::getAsNumber(*this, columnName, value); }
+
+bool Row::get(size_t columnIdx, int64_t& value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, int32_t& value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, int16_t& value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, int8_t&  value) const { return ::getAsNumber(*this, columnIdx, value); }
+
+bool Row::get(std::string const& columnName, int64_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, int32_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, int16_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, int8_t&  value) const { return ::getAsNumber(*this, columnName, value); }
+
+bool Row::get(size_t columnIdx, float&  value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, double& value) const { return ::getAsNumber(*this, columnIdx, value); }
+
+bool Row::get(std::string const& columnName, float&  value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, double& value) const { return ::getAsNumber(*this, columnName, value); }
+
+bool Row::get(size_t columnIdx, bool& value) const {
+    uint8_t number;
+    if (::getAsNumber(*this, columnIdx, number)) {
+        value = number != '0';
+        return true;
+    }
+    return false;
+}
+
+bool Row::get(std::string const& columnName, bool&  value) const {
+    uint8_t number;
+    if (::getAsNumber(*this, columnName, number)) {
+        value = number != '0';
+        return true;
+    }
+    return false;
+}
+
+Row::Cell const& Row::getDataCell(size_t columnIdx) const {
+
+    std::string const context = "Row::getDataCell()  ";
+
+    if (not isValid()) {
+        throw std::logic_error(context + "the object is not valid");
+    }
+    if (columnIdx >= _index2cell.size()) {
+        throw std::invalid_argument(
+                context + "the column index '" + std::to_string(columnIdx) +
+                "'is not in the result set");
+    }
+    return _index2cell.at(columnIdx);
+}
+
+Row::Cell const& Row::getDataCell(std::string const& columnName) const {
+
+    std::string const context = "Row::getDataCell()  ";
+
+    if (not isValid()) {
+        throw std::logic_error(context + "the object is not valid");
+    }
+    if (not _name2indexPtr->count(columnName)) {
```

Comment:
You are doing one search in a map here and one in `at()` below, just use `find()` to get an iterator.

`out_of_range` exception may be better here too.

Reply:
Optimized.